src/features/auth/lib/authMachine.ts

```diff
@@ -50,6 +55,11 @@ const getReferrerID = () => {
 const deleteFarmUrl = () =>
   window.history.pushState({}, "", window.location.pathname);
 
+const guestUser = (): GuestUser => ({
+  type: "GUEST",
+  guestKey: localStorage.getItem(GUEST_KEY),
```

Comment:
Consider moving the local storage interactions into a separate file to keep this file focussed on business logic.

`session.ts` would be a good candidate 👍